src/contracts/core/DelegationManager.sol

```diff
@@ -661,24 +859,52 @@ contract DelegationManager is
         address operator,
         address withdrawer,
         IStrategy[] memory strategies,
-        uint256[] memory shares
+        uint256[] memory sharesToWithdraw,
+        uint64[] memory totalMagnitudes
     ) internal returns (bytes32) {
         require(
             staker != address(0), "DelegationManager._removeSharesAndQueueWithdrawal: staker cannot be zero address"
         );
         require(strategies.length != 0, "DelegationManager._removeSharesAndQueueWithdrawal: strategies cannot be empty");
 
+        uint256[] memory scaledStakerShares = new uint256[](strategies.length);
+
         // Remove shares from staker and operator
         // Each of these operations fail if we attempt to remove more shares than exist
         for (uint256 i = 0; i < strategies.length;) {
+            // check sharesToWithdraw is valid
+            // TODO maybe have a getter to get totalShares for all strategies, like getDelegatableShares
+            // but for inputted strategies
+            uint256 totalShares;
+            if (strategies[i] == beaconChainETHStrategy) {
+                int256 podShares = eigenPodManager.podOwnerShares(staker);
+                totalShares = podShares <= 0 ? 0 : uint256(podShares);
+            } else {
+                totalShares = strategyManager.stakerStrategyShares(staker, strategies[i]);
+            }
+            require(
+                sharesToWithdraw[i] <= _getWithdrawableShares(staker, strategies[i], totalShares, totalMagnitudes[i]),
```

Comment:
perhaps we should pass in `stakerScalingFactor` here instead of loading twice? not a big deal

Reply:
Done, makes more sense after putting things in the SlashingLib https://github.com/Layr-Labs/eigenlayer-contracts/pull/679/commits/a70ed62f13fa8be58c456c8d518ec71e7a8482ab